pkg/admission/decode.go

```diff
@@ -54,3 +55,31 @@ func DecodeCloudProfileConfig(decoder runtime.Decoder, config *runtime.RawExtens
 
 	return cloudProfileConfig, nil
 }
+
+// DecodeBackupBucketConfig decodes the `BackupBucketConfig` from the given `RawExtension`.
+func DecodeBackupBucketConfig(decoder runtime.Decoder, config *runtime.RawExtension) (*gcp.BackupBucketConfig, error) {
+	if config == nil {
+		return nil, nil
+	}
+
+	backupBucketConfig := &gcp.BackupBucketConfig{}
+	if err := util.Decode(decoder, config.Raw, backupBucketConfig); err != nil {
+		return nil, err
+	}
+
+	return backupBucketConfig, nil
+}
+
+// DecodeSeedBackupBucketConfig decodes the `BackupBucketConfig` from the given `SeedBackup`.
+func DecodeSeedBackupBucketConfig(decoder runtime.Decoder, backup *core.SeedBackup) (*gcp.BackupBucketConfig, error) {
+	if backup == nil || backup.ProviderConfig == nil {
+		return nil, nil
+	}
+
+	backupBucketConfig := &gcp.BackupBucketConfig{}
+	if err := util.Decode(decoder, backup.ProviderConfig.Raw, backupBucketConfig); err != nil {
+		return nil, err
+	}
+
+	return backupBucketConfig, nil
```

Comment:
You can call `DecodeBackupBucketConfig` instead.

Reply:
I have removed func `DecodeSeedBackupBucketConfig`, now we are using `DecodeBackupBucketConfig` only